### ISSUE TITLE
Doesn't work with current pdm

### ISSUE THREAD

**Issue by mattiadg** [EmbarkStudios/pdm-plugin-torch#12]
Hi, 
thank you for developing this useful plugin. 
I was wondering if there are plans to make it work with the current version of pdm. I also tried using the code from the open PR but still getting `missing required argument config_prefix`.

I would be happy to help if it would be useful.

**Comment by tgolsson** [comment]
Hey @mattiadg! I did a PR for this pre-vacation but haven't merged it yet. I'll try to get a release out this week! #11. Of course, 2.8 is out since then, but I think the specific one you see is fixed...

**Comment by danielphan2003** [comment]
I tried v23.2.0 on PDM v2.7.4 and it works, but anything later than that breaks with this error:
<details><summary>pdm.exceptions.CandidateNotFound: No candidate is found for `torch` that matches the environment or hashes</summary>
<pre><code>&lt;omitted for brevity&gt;
unearth.evaluator: Skipping link <Link https://files.pythonhosted.org/packages/9c/1e/0da488a46027e2fb671feeed4944b9783905d6fd705069e4c69f03e6ab48/torch-2.1.0-cp39-none-macosx_10_9_x86_64.whl (from https://pypi.org/simple/torch/)>: none of the wheel tags(cp39-none-macosx_10_9_x86_64) are compatible
unearth.evaluator: Skipping link <Link https://files.pythonhosted.org/packages/4c/53/f1e58e147df8601c963df4b15045631f7e3d3caa5973bdf4e54a5cf6834e/torch-2.1.0-cp39-none-macosx_11_0_arm64.whl (from https://pypi.org/simple/torch/)>: none of the wheel tags(cp39-none-macosx_11_0_arm64) are compatible
unearth.evaluator: Skipping package Package(name='torch', version='2.1.0'): version doesn't match ==2.0.1+cu118
unearth.evaluator: Skipping package Package(name='torch', version='2.0.1'): version doesn't match ==2.0.1+cu118
unearth.evaluator: Skipping package Package(name='torch', version='2.0.0'): version doesn't match ==2.0.1+cu118
unearth.evaluator: Skipping package Package(name='torch', version='1.13.1'): version doesn't match ==2.0.1+cu118
unearth.evaluator: Skipping package Package(name='torch', version='1.13.0'): version doesn't match ==2.0.1+cu118
pdm.termui: Error occurs: 
Traceback (most recent call last):
  File "/usr/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/danie/.local/pipx/venvs/pdm/lib/python3.11/site-packages/pdm/installers/synchronizers.py", line 304, in update_candidate
    self.manager.install(can)
  File "/home/danie/.local/pipx/venvs/pdm/lib/python3.11/site-packages/pdm/installers/manager.py", line 33, in install
    installer(str(prepared.build()), self.environment, prepared.direct_url())
                  ^^^^^^^^^^^^^^^^
  File "/home/danie/.local/pipx/venvs/pdm/lib/python3.11/site-packages/pdm/models/candidates.py", line 381, in build
    self.obtain(allow_all=False)
  File "/home/danie/.local/pipx/venvs/pdm/lib/python3.11/site-packages/pdm/models/candidates.py", line 422, in obtain
    raise CandidateNotFound(
pdm.exceptions.CandidateNotFound: No candidate is found for `torch` that matches the environment or hashes
pdm.termui: Error occurs
Traceback (most recent call last):
  File "/home/danie/.local/pipx/venvs/pdm/lib/python3.11/site-packages/pdm/termui.py", line 239, in logging
    yield logger
  File "/home/danie/.local/pipx/venvs/pdm/lib/python3.11/site-packages/pdm_plugin_torch/main.py", line 257, in do_sync
    handler.synchronize()
  File "/home/danie/.local/pipx/venvs/pdm/lib/python3.11/site-packages/pdm/installers/synchronizers.py", line 451, in synchronize
    raise InstallationError("Some package operations are not complete yet")
pdm.exceptions.InstallationError: Some package operations are not complete yet
</code></pre>
</details>

**Comment by tgolsson** [comment]
Hmm. Can you share the full config you're using?

**Comment by danielphan2003** [comment]
For reproducing the error, I'm using following PDM's [other installation methods][other-installation-methods] via pyprojectx.

Here is my pyproject.toml:
```toml
[tool.pdm]

[project]
name = ""
version = ""
description = ""
authors = [
    {name = "danielphan2003"},
    {name = "Daniel Phan", email = "danielphan.2003@gmail.com"},
]
dependencies = []
requires-python = ">=3.11"
license = {text = "MIT"}
readme = "README.md"

[tool.pdm.scripts]
post_install = "pdm plugin add pdm-plugin-torch==23.2.0"

[tool.pdm.plugins.torch]
dependencies = [
   "torch==2.0.1"
]
enable-cpu = true

enable-cuda = true
cuda-versions = ["cu118"]


[tool.pyprojectx]
pdm = "pdm==2.9.3"

# commonly used tools
#black = "black"
#isort = "isort"
#pylint = "pylint"
#pre-commit = "pre-commit"
#jupyter = ["jupyter", "."]

[tool.pyprojectx.aliases]
install = "pdm install"
run = "pdm run"
outdated = "pdm update --outdated"
test = "pdm run pytest"

# example aliases
#clean = "rm -rf .venv .pytest_cache dist"
#clean-all = "pw@ clean && rm -rf .pyprojectx"
#black = "black src tests"
#isort = "isort src tests"
#check-pylint = "pylint src tests"
#check-black = "black src tests --check"
#check = "pw@ check-black && pw@ check-pylint && pw@ test"
#build = "pw@ install && pw@ check && pw@ pdm build"
#notebook = "jupyter notebook -y"

[tool.pyprojectx.os.win.aliases]
# example os specific aliases
#clean = "rmdir /s/q .venv .pytest_cache dist"
#clean-all = "pw@ clean & rmdir /s/q .pyprojectx"
```

My exact commands were:
```bash
curl -LO https://github.com/pyprojectx/pyprojectx/releases/latest/download/wrappers.zip && unzip wrappers.zip && rm -f wrappers.zip
./pw --init pdm
./pw pdm install
./pw pdm torch lock
./pw pdm torch install cu118
```

[other-installation-methods]: https://pdm.fming.dev/latest/#other-installation-methods

**Comment by tgolsson** [comment]
Thanks, added it to our backlog and will try to look soon. I'm eyeing https://github.com/pdm-project/pdm/pull/2102 if it stopped working between 2.7.4 and 2.8.0, but there might be some changed API we're now misusing.

**Comment by tgolsson** [comment]
This is ~sort of fixed in the next release, hopefully made this afternoon. Unfortunately the fix doesn't work due to https://github.com/pdm-project/pdm/pull/2151 only being available in 2.9, so PDM 2.8 can't be supported without a lot of work.
